### ISSUE TITLE
Incorrect printing of return value from functions (specially for strings)

### ISSUE THREAD

**Issue by sanbee** [sanbee/fussy#16]
The following fussy code should print the value of the argument supplied to the function f.

```
f(x){return x;}
f("t")
```

However it prints garbage when the argument is a quoted-string. It prints the format string "e.g. "%10.5f") when supplied a number (e.g. 10pm1).  This behaviour is different when compiled with G++ 4.x, 5.x and 7.x.  With 7.x, print of the numbers work fine. Printing of a string argument does not work with any compiler version.  

The likely place for a bug is in ret() or printcode() functions in vcode.cc

The VM code for this is as follows:

```
0 CONST 1
1 CONST 0
2 rvpush
3 LSYM <TBD>
4 ret
5 STOP
6 vpush
7 "t"
8 vpush
9 f
10 cpush
11 (1 +/- 0)
12 call
13 printcode
14 STOP
```

**Comment by sanbee** [comment]
This is the same as Issue #3.

**Comment by sanbee** [comment]
Fixes from ErrObjStr branch for this issue has now been merged into the master branch.  That fixes this and issue #3. 

The bug was due to the use of C++ exception mechanism for servicing the ret and break VM instructions.  ret() and break_code() calls in the VM code never returned (since they threw ReturnException and BreakException to transfer control flow in the VM execution).  This left the type field of the symbol on the VM stack in a garbled state.  Printing of the return values from a function was therefore unpredictable (ranged from garbage output to seg fault).

Use of C++ exception mechanism as a way to manage VM call-stack has now been replaced with use of state variables to achieve the same result.  This essentially also fixes these (and a few other issues).
